### ISSUE TITLE
stdstar job "succeeds" even if some petals fail

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#1781]
stdstar jobs exit with code=0 (success) even if individual petals fail their processing.  e.g. from /global/cfs/cdirs/desi/spectro/redux/daily/run/scripts/night/20220527/stdstarfit-20220527-00136980-a0123456789-59579365.log:
```
...
CRITICAL:util.py:149:runcmd: FAILED rank 0 exception
...
ERROR:proc_joint_fit.py:543:main: stdstars.main failed for stdstars-3-00136980.fits.gz
...
ERROR:proc_joint_fit.py:555:main: 3/30 stdstar commands failed
...
INFO:proc_joint_fit.py:629:main: All done at Sat May 28 01:04:32 2022
SUCCESS: done at Sat May 28 01:04:33 PDT 2022
```
`desi_proc_joint_fit` needs to intercept these individual failures and exit with an overall failure code instead of success.

**Comment by sbailey** [comment]
The logic appears to be purposeful, so documenting it for discussion before overriding:
From [proc_joint_fit.py L554-561](https://github.com/desihub/desispec/blob/b7fbeba8a8bb518d814e887da1e648ba669143b9/py/desispec/scripts/proc_joint_fit.py#L554-L561):
```python
        if rank == 0 and num_err > 0:
            log.error(f'{num_err}/{num_cmd} stdstar commands failed')

        sys.stdout.flush()
        if num_err>0 and num_err==num_cmd:
            if rank == 0:
                log.critical('All stdstar commands failed')
            sys.exit(1)
```

i.e. individual failures log an error, but it requires all petals to fail before the command itself exits with an error thus triggering the job to fail and canceling dependencies.  I suspect this was originally motivated by wanting the daily pipeline to proceed with whatever it could.  In practice we've been slowly undoing that choice in other cases and I suspect we want to do that here as well.

**Comment by sbailey** [comment]
Implemented in PR #1820; closing.
